### PR TITLE
Update and clean up `pyproject.toml`.

### PR DESCRIPTION
**Pull request by real-yfprojects**
* Increase min py version to 3.7

* Clean up pyproject.toml and update it from setuptools config

* Remove setuptools config

* Run ci tests on PR.
* Run tests on py3.10 and py3.11 also.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
 classifiers = [
     'Intended Audience :: Developers',
     'Topic :: Software Development :: Documentation',
     'Topic :: Software Development :: Quality Assurance',
     'License :: OSI Approved :: MIT License',
-    'Programming Language :: Python :: 3.9',
-    'Programming Language :: Python :: 3.8',
-    'Programming Language :: Python :: 3.7',
-    'Programming Language :: Python :: 3.6',
+    'Programming Language :: Python :: 3 :: Only',
```
Comment (akaihola) [inline pyproject.toml:23]: I think it would be good to explicitly list the supported Python versions here so they show up on the PyPI page.

**Review by akaihola — CHANGES_REQUESTED** [review]
Let's do these ones in 1.8.3 as well so the first 1.8.2 release has as few changes as possible compared to `darglint` 1.8.1.

File: pyproject.toml [hunk 2]
```diff
@@ -1,40 +1,49 @@
...
 keywords = ["documentation", "linter", "development"]
 classifiers = [
     'Intended Audience :: Developers',
     'Topic :: Software Development :: Documentation',
     'Topic :: Software Development :: Quality Assurance',
     'License :: OSI Approved :: MIT License',
-    'Programming Language :: Python :: 3.9',
-    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3 :: Only',
```
Comment (real-yfprojects) [inline pyproject.toml:24]: Is l.23 needed?

Comment (akaihola) [reply pyproject.toml:24]: Sorry, what's `l.23`?

Comment (akaihola) [reply pyproject.toml:24]: Ah, line 23. I'd need to check the documentation for classifiers, I don't know.

Comment (akaihola) [reply pyproject.toml:24]: I checked ~15 of the first hits for the [PyPI search on `Python :: 3 :: Only`, and about half of those seem to also include `Python :: 3`. Based on that, I'd say this is purely a matter of taste.

Comment (real-yfprojects) [reply pyproject.toml:24]: will do

**Review by akaihola — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (real-yfprojects) [reply pyproject.toml:24]: I couldn't find any specific documentation of the classifiers and how parenting works. Pypis filtering by classifier doesn't work, so I couldn't check through that.

**Comment by real-yfprojects** [comment]
Do you prefer a linear git history or merge commits?

File: pyproject.toml (hunk 2 above)
Comment (akaihola) [reply pyproject.toml:24]: I'm happy with this as it is now.

**Comment by akaihola** [comment]
I prefer merge commits after rebasing on `main`.

**Comment by real-yfprojects** [comment]
Let's configure the third option as described [here](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-merging-for-pull-requests):

default-commit-message-dropdown

**Comment by akaihola** [comment]
@real-yfprojects, we now have
kuva

Is that default for the commit message useful in your experience?

**Comment by real-yfprojects** [comment]
> Is that default for the commit message useful in your experience?

:+1:

**Pull request merged**